data-generation/src/main/scala/thesis/Landy.scala

```diff
@@ -12,28 +12,45 @@ import scala.collection.mutable
 import scala.math.Ordered.orderingToOrdered
 
 
-class Landy(graph: LandyAttributeGraph) extends TemporalGraph[LandyEntityPayload, LandyEntityPayload] {
-  override val vertices: VertexRDD[LandyEntityPayload] = graph.vertices
-  override val edges: EdgeRDD[LandyEntityPayload] = graph.edges
-  override val triplets: RDD[EdgeTriplet[LandyEntityPayload, LandyEntityPayload]] = graph.triplets
+class Landy(graph: LandyAttributeGraph) extends TemporalGraph {
+  val vertices: VertexRDD[LandyEntityPayload] = graph.vertices
+  val edges: EdgeRDD[LandyEntityPayload] = graph.edges
 
-  override def snapshotAtTime(instant: Instant): Graph[LandyEntityPayload, LandyEntityPayload] = {
+  def snapshotAtTimeLandy(instant: Instant): Graph[LandyEntityPayload, LandyEntityPayload] = {
     val vertices = localVertices
       .filter(vertex =>
         vertex._2.validFrom < instant &&
           vertex._2.validTo >= instant
       )
       .map(vertex => (vertex._2.id, vertex._2))
 
-    val edges = this.edges.filter(edge =>
+    val edges = this.graph.edges.filter(edge =>
       edge.attr.validFrom < instant &&
         edge.attr.validTo >= instant
     )
     Graph(vertices, edges)
   }
 
+  override def snapshotAtTime(instant: Instant): Snapshot = {
+    val vertices = localVertices
+      .filter(vertex =>
+        vertex._2.validFrom < instant &&
+          vertex._2.validTo >= instant
+      )
+      .map(vertex => (vertex._2.id, vertex._2))
+      .mapValues(_.attributes)
+
+    val edges = this.graph.edges.filter(edge =>
+      edge.attr.validFrom < instant &&
+        edge.attr.validTo >= instant
```

Comment:
```suggestion
      edge.attr.interrval.contains(instant)
```

Reply:
Dette er vel bedre for en annen PR? Eller skal jeg gjøre en commit som endrer implementasjonen av .contains() og bruke den rundt omkring

Reply:
Tenker du kan bruke contains() her, og så i en annen PR kan vi endre på den og sørge for at ting blir rett. Siden contains her blir jo _nesten_ rett hvertfall

Reply:
ja, jeg bare copyPastet fra den andre implementasjonen. Jeg kan endre 👍